core/unit_test/tools/TestEventCorrectness.hpp

```diff
@@ -55,9 +55,9 @@ class Serial;
 class OpenMP;
 class Cuda;
 class Threads;
+class HIP;
 namespace Experimental {
 class SYCL;
-class HIP;
```

Comment:
This code is not invalid. The compiler compiles that `Experimental::HIP` is already defined. I don't think that many users will hit that problem and the fix in user code is pretty simple.

Reply:
not -> now

In other words, this PR would break code that had forward declarations of `HIP` or `HIPSpace` in `Kokkos::Experimental` namespace.